packages/glimmer-runtime/lib/compiled/opcodes/content.ts

```diff
@@ -27,12 +39,20 @@ export class AppendOpcode extends Opcode {
   evaluate(vm: VM) {
     let reference = vm.frame.getOperand();
 
-    let mapped = map(reference, normalizeTextValue);
+    let mapped = map(reference, normalizeTextOrTrustedValue);
     let cache = new ReferenceCache(mapped);
-    let node = vm.stack().appendText(cache.peek());
-
-    if (!isConst(reference)) {
-      vm.updateWith(new UpdateAppendOpcode(cache, node));
+    let value = cache.peek();
+
+    if (isSafeString(value)) {
+      let bounds = vm.stack().insertHTMLBefore(null, value);
```

Comment:
Simple helpers are required to be a pure function of its inputs - when do you expect that helper to recompute? (Class based helpers on the other hand are explicitly designed for that scenario)